### PR TITLE
Lay the ground to business-logic types and a usysconf binary specifically for Solus

### PR DESCRIPTION
**Pull request by livingsilver94**
It's not that useful, more of a guideline.

### CODE REVIEW

**Comment by livingsilver94** [comment]
Converting to draft since we're switching to YAML.

File: src/bin/usysconf/cli/mod.rs
Comment (tarkah) [inline src/bin/usysconf/cli/mod.rs:1]: Personal preference, but I'd prefer to keep `main.rs` next to `lib.rs` in the root `src` folder. `bin` feels like it should be used for ancillary binaries and just adds extra nesting here.

Comment (tarkah) [reply src/bin/usysconf/cli/mod.rs:1]: Alternatively we can have separate crates and put all the library code under a crate called `core` or something.

Comment (livingsilver94) [reply src/bin/usysconf/cli/mod.rs:1]: It *is* an ancillary binary. AFAIU usysconf will be a library, the binary is only for backward compatibility in Solus. See #2.

Comment (tarkah) [reply src/bin/usysconf/cli/mod.rs:1]: Ahh, that makes perfect sense then 👍

File: src/trigger.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            wait_list.push(t.run()?.status());
+        }
+        let results = futures::prelude::future::try_join_all(wait_list).await?;
+        for r in results {
+            if !r.success() {
+                return Err(Error::TaskFailed(r));
+            }
+        }
+        Ok(())
+    }
```
Comment (tarkah) [inline src/trigger.rs:92]: You can eliminate the allocation and improve control flow by:

```rs
future::try_join_all(self.tasks.iter().map(|t| async move {
    let mut handle = t.run()?;
    let status = handle.status().await?;

    if !status.success() {
        Err(Error::TaskFailed(status))
    } else {
        Ok(())
    }
}))?;

Ok(())
```

**Comment by livingsilver94** [comment]
I'll stop here for this PR. Ready to be reviewed and merged.

**Comment by ikeycode** [comment]
lets go! Ty ty ty!

**Pull request merged**